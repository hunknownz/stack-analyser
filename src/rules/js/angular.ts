import { register } from '../../rules.js';

register({
  tech: 'angular',
  files: ['angular.json'],
  dependencies: [
    { type: 'npm', name: 'angular' },
    { type: 'npm', name: /^@angular\//, example: '@angular/forms' },
  ],
});
