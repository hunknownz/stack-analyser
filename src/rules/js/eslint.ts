import { register } from '../../register.js';

register({
  tech: 'eslint',
  name: 'Eslint',
  type: 'tool',
  files: [
    '.eslintrc',
    '.eslintrc.cjs',
    '.eslintrc.json',
    '.eslintrc.js',
    'eslint.config.js',
    'eslint.config.cjs',
    'eslint.config.mjs',
  ],
  dependencies: [{ type: 'npm', name: 'eslint' }],
});
