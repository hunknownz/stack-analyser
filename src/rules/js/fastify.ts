import { register } from '../../register.js';

register({
  tech: 'fastify',
  name: 'Fastify',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'fastify' }],
});
