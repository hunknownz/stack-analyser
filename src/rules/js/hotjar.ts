import { register } from '../../register.js';

register({
  tech: 'hotjar',
  name: 'Hotjar',
  type: 'tool',
  dependencies: [{ type: 'npm', name: '@hotjar/browser' }],
});
