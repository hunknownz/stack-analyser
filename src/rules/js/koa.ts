import { register } from '../../register.js';

register({
  tech: 'koa',
  name: 'Koa',
  type: 'tool',
  dependencies: [{ type: 'npm', name: 'koa' }],
});
