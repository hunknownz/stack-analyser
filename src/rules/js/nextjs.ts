import { register } from '../../register.js';

register({
  tech: 'nextjs',
  name: 'Next.js',
  type: 'framework',
  files: ['next.config.js'],
  dependencies: [{ type: 'npm', name: 'next' }],
});
