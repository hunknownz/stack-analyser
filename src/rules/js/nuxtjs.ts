import { register } from '../../register.js';

register({
  tech: 'nuxtjs',
  name: 'Nuxt.js',
  type: 'framework',
  files: ['nuxt.config.js'],
  dependencies: [{ type: 'npm', name: 'nuxt' }],
});
