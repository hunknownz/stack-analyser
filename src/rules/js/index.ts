import './angular.js';
import './bootstrap.js';
import './esbuild.js';
import './eslint.js';
import './express.js';
import './fastify.js';
import './hotjar.js';
import './koa.js';
import './prettier.js';
import './prisma.js';
import './react.js';
import './reactEmail.js';
import './rollup.js';
import './sequelize.js';
import './socketio.js';
import './storybook.js';
import './strapi.js';
import './stylelint.js';
import './tailwind.js';
import './typescript.js';
import './vite.js';
import './vue.js';
import './webpack.js';
