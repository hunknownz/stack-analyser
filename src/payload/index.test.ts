import { describe, expect, it, vi } from 'vitest';

import { Payload } from './index.js';

let id = 0;
vi.mock('../common/nid.ts', () => {
  return { nid: () => `${id++}` };
});

describe('Payload', () => {
  it('should detect nested dependencies', () => {
    const root = new Payload({ name: 'root', folderPath: '/' });
    root.addChild(
      new Payload({ name: 'PubSub', folderPath: '/', tech: 'gcp.pubsub' })
    );

    expect(root.toJson()).toMatchSnapshot();
  });
});
