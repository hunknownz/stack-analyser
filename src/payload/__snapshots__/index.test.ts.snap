// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Payload > should detect nested dependencies 1`] = `
{
  "childs": [
    {
      "childs": [],
      "dependencies": [],
      "edges": [],
      "group": "hosting",
      "id": "2",
      "inComponent": null,
      "languages": {},
      "name": "GCP",
      "path": [
        "/",
      ],
      "tech": "gcp",
      "techs": [],
    },
    {
      "childs": [],
      "dependencies": [],
      "edges": [],
      "group": "component",
      "id": "1",
      "inComponent": "2",
      "languages": {},
      "name": "PubSub",
      "path": [
        "/",
      ],
      "tech": "gcp.pubsub",
      "techs": [],
    },
  ],
  "dependencies": [],
  "edges": [],
  "group": "component",
  "id": "0",
  "inComponent": null,
  "languages": {},
  "name": "root",
  "path": [
    "/",
  ],
  "tech": null,
  "techs": [],
}
`;
